types/react-bootstrap-table/index.d.ts

```diff
@@ -1,66 +1,207 @@
-// Type definitions for react-bootstrap-table 2.6
+// Type definitions for react-bootstrap-table 4.0
 // Project: https://github.com/AllenFang/react-bootstrap-table
-// Definitions by: Frank Laub <https://github.com/flaub>, Aleksander Lode <https://github.com/alelode>, Josué Us <https://github.com/UJosue10>
+// Definitions by: Frank Laub <https://github.com/flaub>,
+//                 Aleksander Lode <https://github.com/alelode>,
+//                 Josué Us <https://github.com/UJosue10>
+//                 Janeene Beeforth <https://github.com/dawnmist>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.3
 
-/// <reference types="node" />
-
 // documentation taken from http://allenfang.github.io/react-bootstrap-table/docs.html
 
-import { ComponentClass, Props, ReactElement } from 'react';
-import { EventEmitter } from 'events';
+import { Component, CSSProperties, Props, ReactElement, SyntheticEvent } from 'react';
+
+/**
+ * Row Keys type. Used as the unique primary index for a row.
+ * This should be either a string or a number.
+ */
+export type RowKey<T extends number | string> = T;
```

Comment:
This is only ever used below as `RowKey<any>`, so the constraint is effectively lost and `any` is used for those types instead ([example](https://www.typescriptlang.org/play/#src=type%20RowKey%3CT%20extends%20string%20%7C%20number%3E%20%3D%20T%3B%0D%0Aconst%20x%3A%20RowKey%3Cany%3E%20%3D%20true%3B)). If you want to restrict row keys to `string | number`, you need to define the type parameter and constraint on the various types or declarations below that use `RowKey`.